src/mailimf_write_generic.rs

```diff
@@ -1581,18 +1581,14 @@ unsafe fn mailimf_date_time_write_driver(
     mut col: *mut libc::c_int,
     mut date_time: *mut mailimf_date_time,
 ) -> libc::c_int {
-    let mut r: libc::c_int = 0;
-    let mut date_str: [libc::c_char; 256] = [0; 256];
-    let mut wday: libc::c_int = 0;
-    wday = dayofweek(
+    let wday = dayofweek(
         (*date_time).dt_year,
         (*date_time).dt_month,
         (*date_time).dt_day,
     );
-    snprintf(
-        date_str.as_mut_ptr(),
-        256i32 as libc::size_t,
-        b"%s, %i %s %i %02i:%02i:%02i %+05i\x00" as *const u8 as *const libc::c_char,
+
+    let date_str = format!(
+        "{}, {} {} {} {:02}:{:02}:{:02} +{:05}",
```

Comment:
magnus and me think that the plus after the percent sign in `%+05i` always forces a sign before the integer, which may be `+` or `-`. [via](http://www.cplusplus.com/reference/cstdio/printf/)
`+{:05}` always adds `+` literally.